orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -423,10 +423,8 @@ private void addSourceToWorks(OrcidProfile orcidProfile, String amenderOrcid) {
     private void addSourceToWorks(OrcidWorks orcidWorks, String amenderOrcid) {
         if (orcidWorks != null && !orcidWorks.getOrcidWork().isEmpty() && amenderOrcid != null) {
             for (OrcidWork orcidWork : orcidWorks.getOrcidWork()) {
-                if (orcidWork.getSource() == null || StringUtils.isBlank(orcidWork.getSource().retrieveSourcePath())) {
```

Comment:
@nitw-shobhit I'm pretty sure this is going to cause previous valid sources to be overwritten. See OrcidProfileManagerImpl.updateOrcidProfile. @wjrsimpson @amontenegro do you agree?